### PR TITLE
[gtsam] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
Related issue #5728.

No feature need to test.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
It failed on static now. It needs more time to fix this issue. Besides this, this port doesn't currently support ARM and UWP due to it needs to depend on boost.

File: ports/gtsam/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,43 @@
+include(vcpkg_common_functions)
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO borglab/gtsam
+    REF 3.2.3
```
Comment (JackBoosY) [inline ports/gtsam/portfile.cmake]: Use commit ID instead of tag name.

File: ports/gtsam/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,43 @@
+include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/gtsam/portfile.cmake]: Remove this line.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
The failure on `static `is due to this:
```
boost_serialization-vc140-mt-gd.lib(utf8_codecvt_facet.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_iarchive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(extended_type_info.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(archive_exception.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_oarchive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(extended_type_info_typeid.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_oserializer.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_text_oprimitive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(text_oarchive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_iserializer.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_text_iprimitive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(text_iarchive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_archive.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_serialization-vc140-mt-gd.lib(basic_serializer_map.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_filesystem-vc140-mt-gd.lib(path_traits.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_filesystem-vc140-mt-gd.lib(path.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_filesystem-vc140-mt-gd.lib(operations.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_filesystem-vc140-mt-gd.lib(codecvt_error_category.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_filesystem-vc140-mt-gd.lib(windows_file_codecvt.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_timer-vc140-mt-gd.lib(cpu_timer.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
boost_chrono-vc140-mt-gd.lib(chrono.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MTd_StaticDebug' doesn't match value 'MDd_DynamicDebug' in DSFVector.cpp.obj
```

**Comment by NancyLi1013** [comment]
Please feel free to ask us to reopen this PR if you need it.

**Pull request closed without merging**